### PR TITLE
CORTX-31126 FA and RBF of CSM tests with error code & message check enabled

### PR DESCRIPTION
**Pull request by RajeshPratapSInghChouhan**
# Problem Statement
- RBF of CSM tests with error code & message check enabled

# Design
-  We have rest_response_data.yaml, collection of all the error code & messages of REST calls.
- All tests must be referring the same file so in case if there is any change or need any reference, can be updated or looked into a single file.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/csm/rest/test_bucket.py [hunk 1]
```diff
@@ -160,12 +169,10 @@ def test_578(self):
...
                                    const.BAD_REQUEST)
         if CSM_REST_CFG["msg_check"] == "enable":
-            assert_utils.assert_equals(response.json()["error_code"],
-                                    str(resp_error_code))
-            assert_utils.assert_equals(response.json()["message"],
-                                  msg)
-        assert_utils.assert_equals(response.json()["message_id"],
-                                  resp_msg_id)
+            self.log.info("Verifying error response...")
+            assert_utils.assert_equals(response.json()["error_code"], resp_error_code)
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_bucket.py:173]: Can u also log error message  "Error code mismatch" in case of assertion

**Review by kachhwahadivya — APPROVED** [review]

File: tests/csm/rest/test_bucket.py (hunk 1 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_bucket.py:173]: as assert_equals printing both the values, so skipping.

**Comment by RajeshPratapSInghChouhan** [comment]
(virenv) [root@ssc-vm-g2-rhev4-1909 cortx-test]# pytest --local True --collect-only
:::::::::::::::::
Successfully unmounted directory
=============================== 2154 tests collected in 11.35s ===============================

**Comment by RajeshPratapSInghChouhan** [comment]
Jobs triggered:
https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/39/
https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/40/
https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/41/
https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/42/
TE executed :
[TEST-28687](https://jts.seagate.com/browse/TEST-28687)
[TEST-29872](https://jts.seagate.com/browse/TEST-29872)
[TEST-35138](https://jts.seagate.com/browse/TEST-35138)
[TEST-40804](https://jts.seagate.com/browse/TEST-40804)
[TEST-35139](https://jts.seagate.com/browse/TEST-35139)
Results : all expected, for 41 failed test with csm_checks True, have 2 Bugs:
[CORTX-31296](https://jts.seagate.com/browse/CORTX-31296)
[CORTX-31295](https://jts.seagate.com/browse/CORTX-31295)

**Review by ketanarlulkar — APPROVED** [review]

File: tests/csm/rest/test_csm_users.py [hunk 2]
```diff
@@ -236,7 +240,8 @@ def test_28936(self):
                 time.sleep(30)
         if not pod_up:
             assert pod_up, "Pod is not up so cannot proceed. Test Failed"
-        assert response1.status_code == HTTPStatus.SERVICE_UNAVAILABLE.value, "Account creation failed."
+        assert response1.status_code == HTTPStatus.SERVICE_UNAVAILABLE.value, (
```
Comment (aditipande96) [inline tests/csm/rest/test_csm_users.py:243]: This can be HTTPStatus.SERVICE_UNAVAILABLE

File: tests/csm/rest/test_csm_users.py [hunk 3]
```diff
@@ -48,12 +49,14 @@
...
-    """REST API Test cases for CSM users
     """
-
+    REST API Test cases for CSM users
+    """
     @classmethod
     def setup_class(cls):
-        """ This is method is for test suite set-up """
+        """
+        This is method is for test suite set-up
```
Comment (aditipande96) [inline tests/csm/rest/test_csm_users.py:58]: Please correct the docstring

File: tests/csm/rest/test_csm_users.py [hunk 4]
```diff
@@ -406,22 +419,24 @@ def test_5001(self):
...
             expect_status_code=const.BAD_REQUEST,
             sort_by=invalid_sortby,
-            verify_negative_scenario=True)
-        self.log.info("Response : %s", response)
+            return_actual_response=True)
         self.log.info("Verifying the response for invalid value for sort_by")
-        assert response, "Status code check has failed check has failed."
-        self.log.info("Verified the response for invalid value for sort_by")
-
+        assert_utils.assert_equals(response.status_code, const.BAD_REQUEST)
```
Comment (aditipande96) [inline tests/csm/rest/test_csm_users.py:429]: Can this be HTTPStatus.BAD_REQUEST?

File: tests/csm/rest/test_csm_users.py [hunk 5]
```diff
@@ -431,34 +446,32 @@ def test_5001(self):
...
+        resp_msg_index = test_cfg["message_index"]
+        msg = resp_data[resp_msg_index]
         self.log.info("Fetching csm user with empty sort by string...")
         response = self.csm_obj.list_csm_users(
             expect_status_code=const.BAD_REQUEST,
             sort_by="",
             return_actual_response=True)
-
+        self.log.info("Verifying the response status code")
+        assert_utils.assert_equals(response.status_code, const.BAD_REQUEST)
```
Comment (aditipande96) [inline tests/csm/rest/test_csm_users.py:464]: Above comment applicable

File: tests/csm/rest/test_csm_users.py [hunk 6]
```diff
@@ -675,37 +701,36 @@ def test_5005(self):
...
+        msg = resp_data[resp_msg_index]
         self.log.info(
             "Fetching the response for empty sort_by parameter with the expected status code")
         response = self.csm_obj.list_csm_users_empty_param(
             expect_status_code=const.BAD_REQUEST,
             csm_list_user_param="dir",
             return_actual_response=True)
-
+        self.log.info("Verifying the response status code")
+        assert response.status_code, const.BAD_REQUEST
```
Comment (aditipande96) [inline tests/csm/rest/test_csm_users.py:720]: Above comment applicable

File: tests/csm/rest/test_csm_users.py (hunk 2 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_users.py]: done

File: tests/csm/rest/test_csm_users.py (hunk 3 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_users.py]: done

File: tests/csm/rest/test_csm_users.py (hunk 4 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_users.py:428]: will have a ticket to remove where we used commons\constants under our rest api tests

File: tests/csm/rest/test_csm_users.py (hunk 5 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_users.py:463]: will have a ticket to remove where we used commons\constants under our rest api tests

File: tests/csm/rest/test_csm_users.py (hunk 6 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_users.py:719]: will have a ticket to remove where we used commons\constants under our rest api tests

**Review by aditipande96 — APPROVED** [review]

File: config/csm/rest_response_data.yaml [hunk 7]
```diff
@@ -43,17 +44,27 @@
     - "Max_buckets: Not a valid integer."
     - "Suspended: Not a valid boolean."
     - "Op_mask: Invalid value."
+  MalformedRequest:
+    - "{}: Must be one of: {}"
```
Comment (seagate-sarang-sawant) [inline config/csm/rest_response_data.yaml:48]: Start using string template for externalized strings or messages . The references of string format is insecure and comes in critical category.

Comment (RajeshPratapSInghChouhan) [reply config/csm/rest_response_data.yaml]: updated

**Review by seagate-sarang-sawant — APPROVED** [review]
The variable names A or B needs to changed to logical names in string templates in next PR.

**Pull request merged**